### PR TITLE
[APIM] Fix rejected subscription request not being deleted

### PR DESCRIPTION
**Pull request by DDH13**
Related to https://github.com/wso2/api-manager/issues/4980

This pull request introduces improvements and fixes to the subscription workflow handling in the API Manager, particularly around the deletion and update flows for subscriptions with special statuses. It ensures that subscriptions in certain states (like ON_HOLD or REJECTED) are handled more robustly and that workflow states are cleaned up appropriately to avoid stuck or inconsistent states. Additionally, a comprehensive test suite has been added for the subscription creation approval workflow executor.

**Subscription workflow handling improvements:**

* Subscriptions in the `DELETE_PENDING` state now have their pending deletion workflows cleaned up before transitioning to `TIER_UPDATE_PENDING`, preventing workflow state inconsistencies.
* When removing subscriptions with `ON_HOLD` or `REJECTED` status, the system bypasses the approval workflow and deletes the subscription directly, ensuring these do not get stuck in workflow queues. [[1]](diffhunk://#diff-eea96b2f615001930c3a7cf2ec3802f9dc86ac99cabcd2ae10716b934c76c3a2R1804-R1852) [[2]](diffhunk://#diff-eea96b2f615001930c3a7cf2ec3802f9dc86ac99cabcd2ae10716b934c76c3a2R1867-R1883)
* For subscriptions in `TIER_UPDATE_PENDING`, the workflow cleanup process is made more robust, with improved error handling and logging.

**Fixes for legacy/stuck workflow states:**

* When removing a subscription, the code detects and cleans up legacy stuck workflows (e.g., a REJECTED subscription that was incorrectly sent through the delete workflow before the new bypass logic), ensuring these are deleted directly and do not block future operations.

**Testing improvements:**

* Added a new test class `SubscriptionCreationApprovalWorkflowExecutorTest` to verify all major transitions and edge cases in the subscription creation approval workflow, including correct handling of approval, rejection, and error scenarios.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIConsumerImplTest.java [hunk 1]
```diff
@@ -845,6 +845,132 @@ public void testRemoveSubscriptionWhenDeletePending() throws APIManagementExcept
...
+
+        APIConsumerImpl apiConsumer = new APIConsumerImplWrapper(apiMgtDAO, apiPersistenceInstance);
+        apiConsumer.removeSubscription(subscribedAPINew, "org1");
+
+        // Verify the subscription was deleted directly, bypassing the approval workflow
+        Mockito.verify(apiMgtDAO, Mockito.times(1)).removeSubscriptionById(1);
+        // Verify no DELETE_PENDING status update was written (no approval workflow started)
+        Mockito.verify(apiMgtDAO, Mockito.never()).updateSubscriptionStatus(
+                Mockito.anyInt(), Mockito.eq(APIConstants.SubscriptionStatus.DELETE_PENDING));
+    }
```
[the remark this answers is not in the document]
Comment (DDH13) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIConsumerImplTest.java:1017]: The PR does not reroute through Delete Pending, can you check again

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 2]
```diff
@@ -1774,6 +1779,25 @@ public void removeSubscription(Identifier identifier, String userId, int applica
...
+            } else if (APIConstants.SubscriptionStatus.REJECTED.equals(status)) {
+                try {
+                    if (apiIdentifier != null) {
+                        subId = apiMgtDAO.getSubscriptionId(apiIdentifier.getUUID(), applicationId);
+                    } else if (apiProdIdentifier != null) {
+                        subId = apiMgtDAO.getSubscriptionId(apiProdIdentifier.getUUID(), applicationId);
+                    }
+                } catch (APIManagementException ex) {
+                    log.warn("Failed to retrieve subscription id for REJECTED subscription");
+                }
```
[the remark this answers is not in the document]
Comment (DDH13) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:1822]: It is the same thing that's done in the old code also, for unblocked APIs

[the remark this answers is not in the document]
Comment (DDH13) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java:1822]: The early return should be fine since both the cases handled by this are when there was never any subscription in the first place right?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIConsumerImplTest.java [hunk 3]
```diff
@@ -845,6 +845,132 @@ public void testRemoveSubscriptionWhenDeletePending() throws APIManagementExcept
...
+
+    @Test
```
Comment (thivindu) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIConsumerImplTest.java]: Can we add tests to cover scenarios where only two workflows are active? Such as creation and deletion, creation and update, deletion and update
